src/plugins/barista/tea.rs

```diff
@@ -0,0 +1,447 @@
+use std::fmt;
+
+use rand::distributions::{Distribution, Standard};
+use rand::seq::SliceRandom;
+use rand::Rng;
+
+const CHANCE_OF_TEA_ADJ: f64 = 0.25;
+const CHANCE_OF_VESSEL_SIZE: f64 = 0.90;
+const CHANCE_OF_VESSEL_ADJ: f64 = 0.85;
+
+// some teas can only be in a certain set of containers. for instance, hohins
+// will only hold some kind of green tea.
+//
+// also, some adjectives only fit certain teas. example: Kangra => green tea,
+// Irish => black tea
+//
+// Some teas are only good at a certain temperature (cold butter tea is
+// disgusting).
+//
+// because of that, we can't just choose a random tea, adjective, and vessel; we
+// need to choose the tea first, then the vessel and adjective based on what tea
+// was chosen.
+
+// Vessels for tea.
+enum VesselType {
+    Teapot,
+    Mug,
+    Bowl,
+    Samovar,
+    Teacup,
+    Hohin,
+    #[allow(dead_code)]
+    Gaiwan,
+    Shiboridashi,
+}
+
+impl VesselType {
+    fn adjective_choices(&self) -> &[&str] {
+        match *self {
+            VesselType::Teapot => &[
+                "vintage",
+                "silver",
+                "English",
+                "antique silver",
+                "jasperware",
+            ],
+            VesselType::Mug => &[
+                "stoneware",
+                "porcelain",
+                "jasperware",
+                "wooden",
+                "Indian-made clay mug",
+            ],
+            VesselType::Bowl => &[
+                "burl wood-and-silver tea",
+                "Tibetan tea",
+                "Tibetan silver tea",
+            ],
+            VesselType::Samovar => &["antique", "vintage", "brass", "silver"],
+            VesselType::Teacup => &[],
+            VesselType::Hohin => &[],
+            VesselType::Gaiwan => &["porcelain", "Ruyao"],
+            VesselType::Shiboridashi => &["porcelain", "red clay"],
+        }
+    }
+}
+
+impl fmt::Display for VesselType {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match *self {
+            VesselType::Teapot => f.write_str("teapot"),
+            VesselType::Mug => f.write_str("mug"),
+            VesselType::Bowl => f.write_str("bowl"),
+            VesselType::Samovar => f.write_str("samovar"),
+            VesselType::Teacup => f.write_str("teacup"),
+            VesselType::Hohin => f.write_str("hohin"),
+            VesselType::Gaiwan => f.write_str("gaiwan"),
+            VesselType::Shiboridashi => f.write_str("shiboridashi"),
+        }
+    }
+}
+
+// tea adjectives. brands, places, etc.
+//
+// Kangra is a location in India where some kinds of green
+// tea are made.
+enum Adjective {
+    Newmans,
+    Earl,
+    FairTrade,
+    Homemade,
+    #[allow(dead_code)]
```

Comment:
Why leave the dead stuff in?

Reply:
For parity with the other impl - I'd like to keep it up to date.